include/openspace/util/distanceconversion.h

```diff
@@ -132,39 +83,42 @@ DistanceUnits = {
 };
 
-constexpr std::array<std::string_view, static_cast<int>(DistanceUnit::League) + 1>
-DistanceUnitNamesSingular = {
-    DistanceUnitNanometer, DistanceUnitMicrometer, DistanceUnitMillimeter,
-    DistanceUnitCentimeter, DistanceUnitDecimeter, DistanceUnitMeter,
-    DistanceUnitKilometer, DistanceUnitAU, DistanceUnitLighthour,
-    DistanceUnitLightday, DistanceUnitLightmonth, DistanceUnitLightyear,
-    DistanceUnitParsec, DistanceUnitKiloparsec, DistanceUnitMegaparsec,
-    DistanceUnitGigaparsec, DistanceUnitGigalightyear, DistanceUnitThou, DistanceUnitInch,
-    DistanceUnitFoot, DistanceUnitYard, DistanceUnitChain, DistanceUnitFurlong,
-    DistanceUnitMile, DistanceUnitLeague
-};
-
-constexpr std::array<std::string_view, static_cast<int>(DistanceUnit::League) + 1>
-DistanceUnitNamesPlural = {
-    DistanceUnitNanometers, DistanceUnitMicrometers, DistanceUnitMillimeters,
-    DistanceUnitCentimeters, DistanceUnitDecimeters, DistanceUnitMeters,
-    DistanceUnitKilometers, DistanceUnitAUs, DistanceUnitLighthours,
-    DistanceUnitLightdays, DistanceUnitLightmonths, DistanceUnitLightyears,
-    DistanceUnitParsecs, DistanceUnitKiloparsecs, DistanceUnitMegaparsecs,
-    DistanceUnitGigaparsecs, DistanceUnitGigalightyears, DistanceUnitThous,
-    DistanceUnitInches, DistanceUnitFeet, DistanceUnitYards, DistanceUnitChains,
-    DistanceUnitFurlongs, DistanceUnitMile, DistanceUnitLeague
-};
+// Note that the syntax here is required when initializing constexpr std::arrays with structs
+constexpr std::array<DistanceUnitName, static_cast<int>(DistanceUnit::League) + 1>
+DistanceUnitNames {{
+    { "Nanometer", "Nanometers", "nm" },
+    { "Micrometer", "Micrometers", "um" },
+    { "Millimeter", "Millimeters", "mm" },
+    { "Centimeter", "Centimeters", "cm" },
+    { "Decimeter", "Decimeters", "dm" },
+    { "Meter", "Meters", "m" },
+    { "Kilometer", "Kilometers", "km" },
+    { "AU", "AU", "au" },
+    { "Lighthour", "Lighthours", "lh" },
+    { "Lightday", "Lightdays", "ld" },
+    { "Lightmonth", "Lightmonths", "lm" },
+    { "Lightyear", "Lightyears", "ly" },
+    { "Parsec", "Parsecs", "pc" },
+    { "Kiloparsec", "Kiloparsecs", "kpc" },
+    { "Megaparsec", "Megaparsecs", "Mpc" },
+    { "Gigaparsec", "Gigaparsecs", "Gpc" },
+    { "Gigalightyear", "Gigalightyears", "Gly" },
+    { "Thou", "Thou", "th" },
+    { "Inch", "Inches", "in" },
+    { "Foot", "Feet", "ft" },
+    { "Yard", "Yards", "yd" },
+    { "Chain", "Chains", "ch" },
+    { "Furlong", "Furlongs", "fur" },
+    { "Mile", "Miles", "mi" },
+    { "League", "Leagues", "league"}
+}};
 
 constexpr bool isValidDistanceUnitName(std::string_view name) {
-    for (std::string_view val : DistanceUnitNamesSingular) {
-        if (val == name) {
-            return true;
-        }
-    }
-
-    for (std::string_view val : DistanceUnitNamesPlural) {
-        if (val == name) {
+    for (DistanceUnit unit : DistanceUnits) {
```

Comment:
Use std::find_if - maybe more performant?

Reply:
Performance would not be different using `std::find_if`, but maybe readability. 

I think for now I'll leave this as is since I don't want to spend too much time messing with this conversion code file.